File: 389-find-the-difference/389.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        table = Counter(s)

        for c in t:
            if c in table:
                if table[c] == 1:
                    del table[c]
                else:
                    table[c] -= 1
            else:
                return c

File: 389-find-the-difference/389.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> m1(26);
        vector<int> m2(26);
        for (char c : s)
            ++m1[c - 'a'];
        for (char c : t)
            ++m2[c - 'a'];
        for (int i = 0; i < 26; ++i) {
            if (m1[i] != m2[i])
                return i + 'a';
        }
        return s[0];
    }
};

